### DISCUSSION TITLE
Add support for Firestore bundles

### DISCUSSION THREAD

**Discussion by aaronSig** [invertase/react-native-firebase#5440] [Ideas] [answered]
Cloud Firestore now supports generating document bundles that can be loaded into the clients.

Details here:
https://firebase.google.com/docs/firestore/bundles

Is there any work currently underway to implement this in RNFirebase? 

If not I'm happy to help and submit a PR for this.

**Comment by mikehardy** [comment]
It's a feature we'd love to have but there is no progress yet. Do you have any time to work up a PR for it? I could collaborate with you to get it merged, and there is some existing work to base the feature on from the partner project FlutterFire - https://github.com/FirebaseExtended/flutterfire/pull/6037

If not, it may be some time but it will happen, it's an important feature yes

**Comment by kmsbernard** [comment, marked as answer]
If there is no work in progress yet, I'd like to implement the bundle API and make a PR for it. The work in FlutterFire seems very helpful for this.

**Reply by kmsbernard** [reply]
I've opened a draft PR: https://github.com/invertase/react-native-firebase/pull/6199

**Reply by mikehardy** [reply]
Ooooo - that's fantastic. This has been nagging at me - that we did not have it yet, it's a great cost optimization. Thanks for posting a PR!
